docs/src/modules/components/ThemeContext.js

```diff
@@ -191,7 +191,7 @@ export function ThemeProvider(props) {
   }, [calculatedMode]);
 
   useEnhancedEffect(() => {
-    document.body.dir = direction;
+    document.body.setAttribute('dir', direction);
```

Comment:
Maybe we should update https://mui.com/material-ui/customization/right-to-left/#1-set-the-html-direction too. Per https://jakearchibald.com/2024/attributes-vs-properties/ it might be more accurate.